pkg/asset/installconfig/clusterid.go

```diff
@@ -74,6 +75,15 @@ func generateInfraID(base string, maxLen int) string {
 	}
 	base = strings.TrimRight(base, "-")
 
+	// Return the InfraID provided by the "INFRA_ID" environment variable if valid
+	if envInfraID, ok := os.LookupEnv("INFRA_ID"); ok && envInfraID != "" {
+		validInfraIDString := fmt.Sprintf("^%s-([b-df-hj-np-tv-z0-9]){%d}$", base, randomLen)
+		validInfraIDRegex := regexp.MustCompile(validInfraIDString)
+		if validInfraIDRegex.MatchString(envInfraID) {
+			return envInfraID
+		}
+	}
+
 	// add random chars to the end to randomize
 	return fmt.Sprintf("%s-%s", base, utilrand.String(randomLen))
```

Comment:
Not sure you need all the complicated regix logic, if you verify that the infraID is the cluster name followed by ('-') and 5 random alphanumeric chars with no vowels, then you can just have the env var replace the `utilrand.String(randomLen)` And have some validation to make sure it is 5 chars and if not return an error

Reply:
> Not sure you need all the complicated regix logic, if you verify that the infraID is the cluster name followed by ('-') and 5 random alphanumeric chars with no vowels, 

Not sure if I understand what you mean. This is precisely what the regex does. We must verify the 'cluster name' contained in the 'install-config.yaml' with the modifications done by the original generation, matches the 'cluster name' part in the envInfraID or we can end up using an incorrect 'cluster name' (with invalid characters or wrong length) or even a totally different 'cluster name'.  The only part allowed to be modified by the user is the 'random' part of the InfraID and must be provided with the same original rules (5 alphanumeric lowercase chars with no vowels and ).  Besides that I don't think it's a "complicated regex logic" but the opposite, it's the most simple way to do it following the current rules.

>then you can just have the env var replace the `utilrand.String(randomLen)` And have some validation to make sure it is 5 chars and if not return an error

The env var cannot replace the `utilrand.String(randomLen)` only because it contains the cluster name also. Still, verifying the random part is 5 lowercase chars long with no vowels would require a regex anyway.

Reply:
Yes but the user knows the clustername in advance, so I don't see why we should override it (or override it and then validate it is the same).
The problem that this change is supposed to fix is that we will don't know the random part prior to the installation so we can't instruct the users regarding the naming scheme of the VMs they will provision, we can verify anything else.

Plus why no vowels?

Reply:
> Yes but the user knows the clustername in advance, so I don't see why we should override it (or override it and then validate it is the same).
> The problem that this change is supposed to fix is that we will don't know the random part prior to the installation so we can't instruct the users regarding the naming scheme of the VMs they will provision, we can verify anything else.
>
 
Ok, I get it now, so the `OPENSHIFT_INSTALL_INFRA_ID_OVERRIDE` will contain the random part only.

> Plus why no vowels?

Don't know the actual reason this is what `utilrand.String(randomLen)` generates. https://pkg.go.dev/k8s.io/apimachinery/pkg/util/rand#String